beamer/events.py

```diff
@@ -185,6 +191,11 @@ def _fetch_range(
                 self._blocks_to_fetch = max(EventFetcher._MIN_BLOCKS, self._blocks_to_fetch // 2)
             codec = self._contract.web3.codec
             events = _decode_events(logs, codec, self._chain_id, self._event_abis)
+            events.append(
```

Comment:
Shouldn't it be sufficient to have one `LatestBlockUpdated` per `fetch()` call? By having this in `fetch_range()` it is possible to have multiple calls per fetch(). It wouldn't harm logic wise but is less efficient.
I was thinking putting it somewhere around [this line](https://github.com/beamer-bridge/beamer/blob/5148323511c6c5c77e5a8914db683886a3d24027/beamer/events.py#L209)

Reply:
And additionally, whereever you place this call we need to make sure to also handle exceptions during this call. I think currently it would bubble up until the very top of the thread_func. 
We should handle it like the other calls, maybe take a look at https://github.com/beamer-bridge/beamer/blob/5148323511c6c5c77e5a8914db683886a3d24027/beamer/events.py#L191. 

What is important to me is that we guarantee that this event is in the list of events. Otherwise we break an invariant that we always need to have `latest_blocks` set before each iteration over claims and requests.

Reply:
Good point, I fixed this by adding the block data request to the inital block number request. This way we do not change internal state when the request fails/